crates/transaction-pool/src/validate/eth.rs

```diff
@@ -327,7 +333,7 @@ where
 
         let account = match self
             .client
-            .latest()
+            .state_by_block_hash(at)
```

Comment:
I think this is now slightly problematic, because by the time this is executed, the hash could have been reorged and could result in an error